### PR TITLE
Update all the dependencies to the latest version and make relevent changes to fix test cases for python3 branch

### PR DESCRIPTION
**Pull request by Ugtan**
**Summary of the changes made:**
* Update all the dependencies to the latest version.
* Removed staticfiles tag as the staticfiles tag library was deprecated in django 2.1 and removed in django 3. Read [here](https://stackoverflow.com/questions/55929472/django-templatesyntaxerror-staticfiles-is-not-a-registered-tag-library) for more info.
* Use render in place of render_to_response shortcut as the render_to_response shortcut was deprecated in Django 2.0, and removed in Django 3.0.
* Add DEFAULT_AUTOFIELD to settings.py file to specify the type of primary key to use and suppress warning while running the server.
* Fix the app test cases.
* Add a new package webdriver-manager to dependencies which ease the process of manually setting up the geckodriver setup which is necessary to run test cases. The package automatically downloads the firefox version based on the user's firefox version, extracts it and also add the geckodriver to the path variable.
* Replace remaining JavaException to JException.

### CODE REVIEW

**Comment by rtgdk** [comment]
@Ugtan Please keep on committing every day or atleast every alternate day.

**Comment by rtgdk** [comment]
Ping @santiagotorres for review.

**Comment by rtgdk** [comment]
@Ugtan We can have separate PRs for every task of the project. It would be easier to review and maintain.

File: src/app/views.py [hunk 1]
```diff
@@ -1840,22 +1840,17 @@ def post_to_github(request):
...
-    )
+    context = {}
+    return render(request, 'app/404.html', context)
 
 def handler500(request, exception=None):
-    return render_to_response('app/500.html',
-        context_instance = RequestContext(request)
-    )
+    context = {}
+    return render(request, 'app/500.html', context)
```
Comment (rtgdk) [inline src/app/views.py:1856]: @Ugtan Please comment on these changes - either in the code or on github PR? Like why was this changed. Like for this one - `render_to_response` shortcut was deprecated in Django 2.0, and is removed in Django 3.0.

**Review by rtgdk — CHANGES_REQUESTED** [review]
There are some changes that went into master recently. Make sure to pull them in.

File: requirements.txt [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-djangorestframework==3.11.0
+Django==3.2.4
+jpype1==1.3.0
+numpy==1.20.3
+djangorestframework==3.12.2
 bs4==0.0.1
-requests==2.24
+requests==2.25
 lxml==4.6.3
+webdriver-manager==3.4.2
```
[the remark this answers is not in the document]
Comment (Ugtan) [reply requirements.txt:8]: This package is majorly used for the setup of geckodriver that is used while running the test cases. The package automatically downloads the suitable version of geckodriver based on your firefox version(different users can have different version), extracts it and also add the geckodriver to the path variable. So, there is no manual work that is to be done.

File: requirements.txt [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 social-auth-app-django==4.0.0
-python-dotenv==0.14.0
-gevent==20.6.2
-gunicorn==20.0.4
+python-dotenv==0.17.1
+gevent==21.1.2
+gunicorn==20.1.0
 #for testing
 selenium==3.14.1
 django-oauth-toolkit==1.3.2
```
[the remark this answers is not in the document]
Comment (Ugtan) [reply requirements.txt:16]: Will do

File: src/app/templates/400.html [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {% extends 'app/base.html' %}
 {% load static %}
-{% load staticfiles %}
```
[the remark this answers is not in the document]
Comment (Ugtan) [reply src/app/templates/400.html:3]: The staticfiles tag library was deprecated in django 2.1 and removed in django 3. Please have a look at  [this](https://stackoverflow.com/questions/55929472/django-templatesyntaxerror-staticfiles-is-not-a-registered-tag-library) out for more info.

File: src/src/settings.py [hunk 5]
```diff
@@ -215,6 +215,9 @@
     ),
 }
 
+# To avoid unwanted migrations in the future, either explicitly set DEFAULT_AUTO_FIELD to AutoField
+DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
```
[the remark this answers is not in the document]
Comment (Ugtan) [reply src/src/settings.py:219]: We have not explicitly defined primary_key=True in our models, so an implicit primary key will be used. The line is used to specify the type of primary key to use. The change was not necessary but when running the server it was generating a warning so I added it. You can refer [here](https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys) for more info.

File: src/app/views.py (hunk 1 above)
Comment (Ugtan) [reply src/app/views.py:1856]: Alright will do.

**Comment by Ugtan** [comment]
> @Ugtan Please keep on committing every day or atleast every alternate day.

Sure. I would commit every day or at least every alternate day from now.

> @Ugtan We can have separate PRs for every task of the project. It would be easier to review and maintain

Perfect. I will create a new PR for the next task and would also edit the description of this PR to mention every change made. That would be more feasible and easy to review, right? @rtgdk @SantiagoTorres

**Comment by Ugtan** [comment]
@rtgdk @SantiagoTorres I have made the requested changes. please have a look. And, if there's anything left to be done then, lmk and I will do it.

**Review by rtgdk — APPROVED** [review]
@Ugtan LGTM! Can you post the test case status for both app and api after these changes? I wanted to see how many new tests are passing after Django3.0 changes. I'll merge post that.

**Comment by Ugtan** [comment]
@rtgdk I will post the result for the APP test cases after these changes. (API ones were fixed in the Merge APP and API PR)

```
$ python manage.py test app      

Creating test database for alias 'default'...
System check identified no issues (0 silenced).
....WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
03:24:14.128 [main] ERROR org.apache.jena.rdf.model.impl.RDFDefaultErrorHandler - (line 688 column 3): The element type "spdx:Review" must be terminated by the matching end-tag "</spdx:Review>".
...................................s..s.03:24:40.156 [Thread-13] ERROR org.apache.jena.rdf.model.impl.RDFDefaultErrorHandler - (line 688 column 3): The element type "spdx:Review" must be terminated by the matching end-tag "</spdx:Review>".
.........................
.s
----------------------------------------------------------------------
Ran 86 tests in 168.317s

OK (skipped=5)
Destroying test database for alias 'default'...
```

image

**Comment by rtgdk** [comment]
@Ugtan Merging this. Thanks! Please rebase these changes with the other PR and post test results again for both app and api.
